### PR TITLE
Don't crash on no TTS available

### LINKED ISSUE #958: App crashes when no TTS engine present
Definitely an edge case but, the app crashes when android does not have a TTS engine enabled (e.g. the default LineageOS with F-Droid setup).

### PR DESCRIPTION
**Pull request by TheLastProject**
Fixes #958.

I would love your feedback on this.

### CODE REVIEW

File: app/src/main/org/runnerup/workout/feedback/RUTextToSpeech.java [hunk 1]
```diff
@@ -90,8 +90,14 @@ private String getId(String text) {
...
     }
 
+    public Boolean isAvailable() {
+        return textToSpeech != null;
+    }
+
     @SuppressWarnings("UnusedReturnValue")
     int speak(String text, int queueMode, HashMap<String, String> params) {
+        if (!isAvailable())
+            return 0;
```
Comment (gerhardol) [inline app/src/main/org/runnerup/workout/feedback/RUTextToSpeech.java:100]: curly brackets

File: app/src/main/org/runnerup/workout/feedback/RUTextToSpeech.java [hunk 2]
```diff
@@ -135,6 +141,9 @@ int speak(String text, int queueMode, HashMap<String, String> params) {
      */
     private int speakWithMute(String text, int queueMode,
             HashMap<String, String> params) {
+        if (!isAvailable())
+            return TextToSpeech.ERROR;
```
Comment (gerhardol) [inline app/src/main/org/runnerup/workout/feedback/RUTextToSpeech.java:145]: curly brackets

File: app/src/main/org/runnerup/view/StartActivity.java [hunk 3]
```diff
@@ -602,23 +604,45 @@ private Workout prepareWorkout() {
...
-                mGpsStatus.stop(StartActivity.this);
-
-                // unregister receivers
-                unregisterStartEventListener();
-
-                // This will start the advancedWorkoutSpinner!
-                mTracker.setWorkout(prepareWorkout());
-                mTracker.start();
+                if (!mTracker.isComponentConnected(new TrackerTTS().getName())) {
+                    createNewNoTtsAvailableDialog();
```
Comment (gerhardol) [inline app/src/main/org/runnerup/view/StartActivity.java:641]: Why not start directly and put the popup in front without waiting?
As this is handled, the user can make decision to correct this next time instead.

**Review by gerhardol — COMMENTED** [review]
Looks good!
Have not run it, but I would be surprised if it creates problems if a tts is available
Please squash too (I can do that when merging, but better to adjust the messages manually)

File: app/src/main/org/runnerup/view/StartActivity.java (hunk 3 above)
Comment (gerhardol) [reply app/src/main/org/runnerup/view/StartActivity.java:641]: ok

**Comment by TheLastProject** [comment]
Squashed it.

>Have not run it, but I would be surprised if it creates problems if a tts is available

I have tested it both without and after manually installing a TTS and the messages only trigger when there is no TTS available.

**Review by gerhardol — APPROVED** [review]

**Comment by gerhardol** [comment]
Thanks!

**Comment by TheLastProject** [comment]
Thank you for explaining, I'll try to watch out for that if I contribute more in the future :)

**Comment by DIWesser** [comment]
Thanks for sorting this out, y'all.

**Pull request merged**